Shoko.Server/Scheduling/Jobs/Trakt/CheckTraktTokenJob.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Shoko.Server.Scheduling.Acquisition.Attributes;
+using Shoko.Server.Scheduling.Attributes;
+using Shoko.Server.Scheduling.Concurrency;
+using Shoko.Server.Settings;
+using Shoko.Server.Providers.TraktTV;
+
+namespace Shoko.Server.Scheduling.Jobs.Trakt;
+
+[DatabaseRequired]
+[NetworkRequired]
+[DisallowConcurrencyGroup(ConcurrencyGroups.Trakt)]
+[JobKeyGroup(JobKeyGroup.Trakt)]
+public class CheckTraktTokenJob : BaseJob
+{
+    private readonly ISettingsProvider _settingsProvider;
+    private readonly TraktTVHelper _traktHelper;
+
+    public bool ForceRefresh;
+    public override string TypeName => "Check and Refresh Trakt Token";
+    public override string Title => "Checking and Refreshing Trakt Token";
+
+    public override Task Process()
+    {
+        try
+        {
+            _logger.LogInformation("Processing {Job}", nameof(CheckTraktTokenJob));
+            var settings = _settingsProvider.GetSettings();
+            if (!settings.TraktTv.Enabled || string.IsNullOrEmpty(settings.TraktTv.TokenExpirationDate))
+            {
+                _logger.LogInformation("Trakt is not enabled or no token expiration date is set");
+                return Task.CompletedTask;
+            }
+
+            // Convert the Unix timestamp to DateTime
```

Comment:
We should also probably add a check if the date is in the past, and unset the token if it is instead of trying to refresh.

Reply:
If everybody is okay with my reply below, I'll mark this as resolved